bob-common/src/node.rs

```diff
@@ -81,6 +91,19 @@ impl Node {
         self.conn.read().expect("rwlock").clone()
     }
 
+    pub async fn increase_error_and_clear_conn_if_needed(&self) {
```

Comment:
This should is not needed if the counter is tracked by BobClient. 
Instead you need a function, that checks that counter to be greater than `max_sequential_errors` and resets the connection in this case. Check should be done in 2 phases:
```
if <check> {
    <write lock>
     if <check> {
          <reset connection>
     }
}
```